libpimeval/src/pimParamsPerf.cpp

```diff
@@ -903,13 +994,20 @@ pimParamsPerf::getMsRuntimeForRotate(PimCmdEnum cmdType, const pimObjInfo& obj)
     // For every bit: Read row to SA; move SA to R1; Shift R1 by N steps; Move R1 to SA; Write SA to row
     msRuntime = (m_tR + (bitsPerElement + 2) * m_tL + m_tW); // for one pass
     msRuntime *= numPass;
+    mjEnergy = (m_eR + (bitsPerElement + 2) * m_eL) * numPass;
     // boundary handling
-    msRuntime += 2 * getMsRuntimeForBytesTransfer(numRegions * bitsPerElement / 8);
+    pimParamsPerf::perfEnergy perfEnergyBT = getPerfEnergyForBytesTransfer(numRegions * bitsPerElement / 8);
+    msRuntime += 2 * perfEnergyBT.m_msRuntime;
+    mjEnergy += 2 * perfEnergyBT.m_mjEnergy;
     break;
   default:
     msRuntime = 1e10;
+    mjEnergy = 999999999.9; // todo
   }
 
-  return msRuntime;
+  pimParamsPerf::perfEnergy m_perfEnergy;
+  m_perfEnergy.m_mjEnergy = mjEnergy;
+  m_perfEnergy.m_msRuntime = msRuntime;
+  return m_perfEnergy;
```

Comment:
With updated ctor, you can simply do return perfEnergy(msRuntime, mjEnergy); In addition, local variable does not have m_ prefix